### PR TITLE
yajl-ruby update to v1.4.0

### LINKED ISSUE #6965: Update dependency constraint to allow for yajl-ruby v1.4.0
Hey there! :wave:

I noticed that the constraint you have for yajl-ruby doesn't allow for the latest version to be used.

The constraint I found was `~> 1.3.1`, and the latest version available is `1.4.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### PR DESCRIPTION
**Pull request by nickskalkin**
Resolves #6965

Tested on High Sierra, Ruby 2.5.1, all works fine

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -80,7 +80,7 @@ group :jekyll_optional_dependencies do
     gem "pygments.rb", "~> 1.0"
     gem "rdiscount", "~> 2.0"
     gem "redcarpet", "~> 3.2", ">= 3.2.3"
-    gem "yajl-ruby", "~> 1.3.1"
+    gem "yajl-ruby", "~> 1.4.0"
```
Comment (pathawks) [inline Gemfile]: Any reason not to use `"~> 1.3"` ?

**Comment by ashmaroli** [comment]
Please edit `Issue #6965` in the PR body to `Resolves #6965`

File: Gemfile (hunk 1 above)
Comment (nickskalkin) [reply Gemfile]: Just updated as mentioned in #6965

**Comment by nickskalkin** [comment]
@ashmaroli done

File: Gemfile (hunk 1 above)
Comment (ashmaroli) [reply Gemfile]: Using `"~> 1.3"` is better..

**Comment by pathawks** [comment]
@nickskalkin Thanks for opening this PR :beers:

I think, so long as `yajl-ruby` follows semantic versioning, we should change the constraint to `"~> 1.3"`
This will allow any version of `yajl-ruby` from `1.3.0` up to (but not including) `2.0.0`. This means that our users will be able to take advantage of `1.5.0` as soon as it is released, instead of waiting for another PR like this one.

File: Gemfile (hunk 1 above)
Comment (nickskalkin) [reply Gemfile]: Ah, got it, sorry :) Read '1.3.1' instead of '1.3'

**Comment by pathawks** [comment]
If you're interested, you can read all about RubyGems version specifiers here: http://guides.rubygems.org/patterns/#pessimistic-version-constraint

**Comment by pathawks** [comment]
Thanks for the PR :tada:

@jekyllbot: merge +dev

**Pull request merged**